model/src/main/proto/exploration.proto

```diff
@@ -0,0 +1,145 @@
+  string objective = 5;
+  bool correctness_feedback_enabled = 6;
+  string title = 7;
+  string language_code = 8;
+}
+
+// Data for a param change.
+// Maps from: data/src/main/java/org/oppia/data/backends/gae/model/GaeParamChange.kt
+message ParamChange {
+  string generator_id = 1;
+  string name = 2;
+  CustomizationArgs customization_args = 3;
+}
+
+// Data for a param spec.
+// Maps from: data/src/main/java/org/oppia/data/backends/gae/model/GaeParamSpec.kt
+message ParamSpec {
+  string obj_type = 1;
+}
+
+// Data for a single state
+// Maps from: data/src/main/java/org/oppia/data/backends/gae/model/GaeState.kt
+message State {
+  // Mapping from content_id to a VoiceoverMapping
+  map<string, VoiceoverMapping> recorded_voiceovers = 1;
+  SubtitledHtml content = 2;
+  // Mapping from content_id to a TranslationMapping
+  map<string, TranslationMapping>written_translations = 3;
+  repeated ParamChange param_changes = 4;
+  string classifier_model_id = 5;
+  Interaction interaction = 6;
+  bool solicit_answer_details = 7;
+}
+
+// Data for customization args.
+// Maps from: data/src/main/java/org/oppia/data/backends/gae/model/GaeCustomizationArgs.kt
+message CustomizationArgs {
+  bool parse_with_jinja = 1;
+  string value = 2;
+}
+
+// Mapping from language_code to a VoiceOver
+message VoiceoverMapping {
+  map<string, Voiceover> voiceover_mapping = 1;
+}
+
+// Data for a single voiceover
+// Maps from: data/src/main/java/org/oppia/data/backends/gae/model/GaeVoiceover.kt
+message Voiceover {
+  int64 file_size_bytes = 1;
+  bool needs_update = 2;
+  string file_name = 3;
+}
+
+// Data for a single interaction
+// Maps from: data/src/main/java/org/oppia/data/backends/gae/model/GaeInteractionInstance.kt
+message Interaction {
+  string id = 1;
+  repeated AnswerGroup answer_groups = 2;
+  Solution solution = 3;
+  repeated AnswerGroup confirmed_unclassified_answers = 4;
+  Hint hint = 5;
+  Outcome default_outcome = 6;
+  map<string, CustomizationArgs> customization_args = 7;
+}
+
+// Mapping from language_code to Translation
+message TranslationMapping {
+  map<string, Translation> translation_mapping = 1;
+}
+
+// Data for a single translation
+// Maps from: data/src/main/java/org/oppia/data/backends/gae/model/GaeWrittenTranslation.kt
+message Translation {
+  string html = 1;
+  bool needs_update = 2;
+}
+
+// Data for a single answer group
+// Maps from: data/src/main/java/org/oppia/data/backends/gae/model/GaeAnswerGroup.kt
+message AnswerGroup {
+  string tagged_skill_misconception_id = 1;
+  Outcome outcome = 2;
+  repeated RuleSpec rule_specs = 3;
+  repeated TrainingData training_data = 4;
+}
+
+// Data for a single training data instance
+message TrainingData {
+  int32 answer_group_index = 1;
+  repeated string answers = 2;
+}
+
+// Data for a single solution
+// Maps from: data/src/main/java/org/oppia/data/backends/gae/model/GaeSolution.kt
+message Solution {
+  string interaction_id = 1;
+  bool answer_is_exclusive = 2;
```

Comment:
It may be worth dropping a brief comment here explaining what this means. It doesn't seem self-evident to me unlike most other fields in these protos.

Reply:
Done